powermon/ports/__init__.py

```diff
@@ -1,8 +1,8 @@
 import importlib
 import logging
-from mppsolar.ports.serialport import SerialPort
-from mppsolar.ports.testport import TestPort
-from mppsolar.ports.usbport import USBPort
+from powermon.ports.serialport import SerialPort
+from powermon.ports.testport import TestPort
+from powermon.ports.usbport import USBPort
```

Comment:
these are going to be problematic after the other ports are added, both from a memory perspective and a dependancy perspective
Ive been looking at poetry features to minimise dependancies so the modules are only added if the user needs them (especially, DB abd BLE modules)
Some users dont even use MQTT...

Reply:
there currently on 3 ports here and they use the same lib, mppsolar has 9(ish)
this import will fail if any dependant libs are missing, the current ble lib requires compiling
the old code did this and it caused a heap of trouble

Reply:
BLE is bluetooth low energy right? Is it used as a way to communicate with the hardware? I think we could move the importlib call to the specific port that causes problems. If that doesn't solve it we can revert back to dynamic imports.

I think DB output should be kept out of powermon and moved into a layer that listens to the MQTT events. Keep powermon focused on interacting with the hardware

Reply:
> BLE is bluetooth low energy right? Is it used as a way to communicate with the hardware? I think we could move the importlib call to the specific port that causes problems. If that doesn't solve it we can revert back to dynamic imports.

yes. Some BMSs are currently supported via BLE and Im looking to extend the functionality to more BLE devices. I am hoping to move to a different library that doesnt need compiling (future milestone)



> I think DB output should be kept out of powermon and moved into a layer that listens to the MQTT events. Keep powermon focused on interacting with the hardware

I dont disagree at all (and dont use this myself) but some find it useful and have sent PRs (that have been accepted) for this type of feature
In my setup MQTT is on a different host (as my primary inverter location has intermittant network troubles), so I can (logically) see the use of local storage of data (or caching)

Reply:
Is anyone using powermon yet?